src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayer.java

```diff
@@ -221,6 +222,52 @@ public void setTotalExperience(int exp) {
         this.experienceTotal = exp;
     }
 
+    /**
+     * {@link EntityPlayer#addExperienceLevel(int)} and
+     * {@link EntityPlayer#removeExperienceLevel(int)} don't update the total
+     * experience. This recalculates it for plugins to properly make use of it.
+     *
+     * @see #onAddExperienceLevels(int, CallbackInfo)
+     */
+    private void recalculateTotalExperience() {
+        if (!dontRecalculateExperience.get()) {
```

Comment:
since this is an instance variable, regardless whether it's thread local, you should have it qualified with `this`.